### PR TITLE
Remove development and test files from the gem package

### PR DESCRIPTION
**Pull request by orien**
There are several files in the gem package that aren't useful for downstream projects. Removing these files reduces the gem package size from **34K** to **24K**!

<details><summary>gem contents diff</summary>

```diff
< .github/workflows/ci.yml
< .gitignore
  CHANGES.md
< Gemfile
  README.md
< Rakefile
< TODO.md
  cells.gemspec
  lib/cell.rb
  lib/cell/abstract.rb
  lib/cell/builder.rb
  lib/cell/caching.rb
  lib/cell/collection.rb
  lib/cell/concept.rb
  lib/cell/development.rb
  lib/cell/escaped.rb
  lib/cell/inspect.rb
  lib/cell/layout.rb
  lib/cell/option.rb
  lib/cell/partial.rb
  lib/cell/prefixes.rb
  lib/cell/self_contained.rb
  lib/cell/templates.rb
  lib/cell/testing.rb
  lib/cell/twin.rb
  lib/cell/util.rb
  lib/cell/version.rb
  lib/cell/view_model.rb
  lib/cells.rb
  lib/tasks/cells.rake
< test/builder_test.rb
< test/cache_test.rb
< test/cell_benchmark.rb
< test/cell_test.rb
< test/concept_test.rb
< test/context_test.rb
< test/fixtures/bassist/play.erb
< test/fixtures/cell_test/song/show_with_block.erb
< test/fixtures/comment/layout/show.erb
< test/fixtures/comment/show/show.erb
< test/fixtures/concepts/record/views/layout.erb
< test/fixtures/concepts/record/views/show.erb
< test/fixtures/concepts/record/views/song.erb
< test/fixtures/inherit_views_test/popper/tap.erb
< test/fixtures/inherit_views_test/tapper/play.erb
< test/fixtures/inherit_views_test/tapper/tap.erb
< test/fixtures/partial_test/with_partial/show.erb
< test/fixtures/partials/_show.html.erb
< test/fixtures/partials/_show.xml.erb
< test/fixtures/song/ivar.erb
< test/fixtures/song/show.erb
< test/fixtures/song/with_block.erb
< test/fixtures/song/with_erb.erb
< test/fixtures/song/with_html.erb
< test/fixtures/song/with_locals.erb
< test/fixtures/song_with_layout/happy.erb
< test/fixtures/song_with_layout/merry.erb
< test/fixtures/song_with_layout/show.erb
< test/fixtures/song_with_layout/show_with_layout.erb
< test/fixtures/templates_caching_test/song/show.erb
< test/fixtures/url_helper_test/song/edit.erb
< test/fixtures/url_helper_test/song/with_block.erb
< test/fixtures/url_helper_test/song/with_capture.erb
< test/fixtures/url_helper_test/song/with_content_tag.erb
< test/fixtures/url_helper_test/song/with_form_for_block.erb
< test/fixtures/url_helper_test/song/with_link_to.erb
< test/inspect_test.rb
< test/layout_test.rb
< test/partial_test.rb
< test/prefixes_test.rb
< test/property_test.rb
< test/public_test.rb
< test/render_test.rb
< test/templates_test.rb
< test/test_helper.rb
< test/testing_test.rb
< test/twin_test.rb
```

</details>

### CODE REVIEW

File: cells.gemspec [hunk 1]
```diff
@@ -19,8 +19,11 @@ Gem::Specification.new do |spec|
   spec.metadata['source_code_uri'] = spec.homepage
   spec.metadata['wiki_uri'] = "#{spec.homepage}/wiki"
 
-  spec.files         = `git ls-files`.split("\n")
-  spec.test_files    = `git ls-files -- {test}/*`.split("\n")
+  spec.files = Dir.chdir(__dir__) do
+    `git ls-files -z`.split("\x0").reject do |file|
+      file.start_with?(*%w[.git Gemfile Rakefile TODO cells.gemspec test])
```
Comment (yogeshjain999) [inline cells.gemspec:24]: I think we should keep `cells.gemspec`, it helps while debugging.

Comment (orien) [reply cells.gemspec]: Updated in [`1fc116d`](https://github.com/trailblazer/cells/pull/503/commits/1fc116d41e9af5bd4714ade7714d91947da5a7a9).

Comment (orien) [reply cells.gemspec]: @ yogeshjain999 what do you think? Should we merge this in?

Comment (yogeshjain999) [reply cells.gemspec]: Sorry I missed this, LGTM 🍻

Comment (orien) [reply cells.gemspec]: Thanks.

**Pull request merged**